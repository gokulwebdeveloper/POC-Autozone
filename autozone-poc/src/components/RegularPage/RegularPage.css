/* .header-image {
    background-image: url("../../utils/img/bg-img/bg-8.jpg");
} */

.header-image {
  /* background: url() no-repeat center center fixed; */
  /* background-image: url("../../assets/img/bg-img/bg-8.jpg"); */
  /* i had to use content instead of using an background image in background image in background tag. */
  /* content: url(../../assets/img/bg-img/bg-8.jpg); */
  background-size: cover;
  /* height: auto; */
  min-width: 100%;
}

.quote-header {
  display: flex;
}

.quote-icon {
  color: #0315ff;
}

.content-container {
  padding-left: 80px;
  padding-right: 80px;
}

p {
  font-size: 18px;
  font-weight: 300;
  color: #787878;
  line-height: 2;
}

h6 {
  font-size: 18px;
  line-height: 2;
}