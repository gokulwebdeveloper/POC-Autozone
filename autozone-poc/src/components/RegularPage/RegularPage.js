import React, { Fragment } from "react";
import "./RegularPage.css";
import HeaderImage from "../../assets/img/bg-img/bg-8.jpg";

export default function RegularPage() {
  return (
    <Fragment>
      {/* <div className="container-fluid">Aman</div> */}
      {/* <div className="container-fluid main-container">
        <div className="row header-row"> */}
          {/* <div className="col-md-12 px-0 header-image"></div> */}
          <img class="img-fluid header-image" src={HeaderImage} />
        {/* </div> */}
        {/* <img src={"../../assets/img/bg-img/bg-8.jpg"} /> */}
        <div className="row mt-5 justify-content-center content-row">
          <div className="container content-container">
            <h1 className="mb-5">
              Vivamus sed nunc in arcu cursus mollis quis et orci. Interdum et
              malesuada
            </h1>
            <p className="mt-5 mb-5">
              Mauris viverra cursus ante laoreet eleifend. Donec vel fringilla
              ante. Aenean finibus velit id urna vehicula, nec maximus est
              sollicitudin. Praesent at tempus lectus, eleifend blandit felis.
              Fusce augue arcu, consequat a nisl aliquet, consectetur elementum
              turpis. Donec iaculis lobortis nisl, et viverra risus imperdiet
              eu. Etiam mollis posuere elit non sagittis. Lorem ipsum dolor sit
              amet, consectetur adipiscing elit. Nunc quis arcu a magna sodales
              venenatis. Integer non diam sit amet magna luctus mollis ac eu
              nisi. In accumsan tellus ut dapibus blandit.
            </p>
            <blockquote className="mt-5 mb-5 quote-blockquote">
              <h6 className="quote-header">
                <i className="fa fa-quote-left mr-4 quote-icon" aria-hidden="true"></i>
                Quisque sagittis non ex eget vestibulum. Sed nec ultrices dui.
                Cras et sagittis erat. Maecenas pulvinar, turpis in dictum
                tincidunt, dolor nibh lacinia lacus.
              </h6>
              <span className="ml-5">Liam Neeson</span>
            </blockquote>
            <p className="mt-5">
              Praesent ac magna sed massa euismod congue vitae vitae risus.
              Nulla lorem augue, mollis non est et, eleifend elementum ante.
              Nunc id pharetra magna. Praesent vel orci ornare, blandit mi sed,
              aliquet nisi. Class aptent taciti sociosqu ad litora torquent per
              conubia nostra, per inceptos himenaeos.
            </p>
          </div>
        {/* </div> */}
      </div>
    </Fragment>
  );
}
