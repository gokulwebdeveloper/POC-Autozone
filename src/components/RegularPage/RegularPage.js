import React from "react";
import "./RegularPage.css";
import Data from "./MockData/Mock.json";

export default function RegularPage() {
  return (
    <div className="container-fluid main-container">
      <div className="row header-row">
        <img className="img-fluid w-100 header-image" src={Data[0].imgSrc} />
      </div>
      <div className="container justify-content-center my-5 content-container">
        <div className="row content-row">
          <h2 className="mb-5">{Data[1].h2Para}</h2>
          <p className="mb-5">{Data[1].para1}</p>
          <blockquote className="mb-5 quote-blockquote">
            <h6 className="d-flex quote-header">
              <i
                className="fa fa-quote-left mr-4 quote-icon"
                aria-hidden="true"
              ></i>
              {Data[1].h6Para}
            </h6>
            <span className="ml-5">{Data[1].spanText}</span>
          </blockquote>
          <p>{Data[1].para2}</p>
        </div>
      </div>
    </div>
  );
}
