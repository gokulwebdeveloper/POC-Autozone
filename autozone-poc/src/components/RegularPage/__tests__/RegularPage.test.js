import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import RegularPage from "../RegularPage";
import HeaderImage from "../../../assets/img/bg-img/bg-8.jpg";

let wrapper;
//mount component for testing
beforeEach(() => {
  wrapper = shallow(<RegularPage />);
});

//Cleanup after test execution
afterEach(() => {
  wrapper.unmount();
});

describe("test RegularPage", () => {
  //test if RegularPage component exists
  it("RegularPage Exists", () => {
    expect(wrapper.exists()).toBeTruthy();
  });

  it("should check for the h2 tag", () => {
    const h2Tag = wrapper.find("h2");
    expect(h2Tag.text()).not.toBe("");
  });

  it("should have 2 paragraph tags", () => {
    const p = wrapper.find("p");
    expect(p).toHaveLength(2);
  });

  //SnapShot Match test
  it("SnapShot Match", () => {
    const page = renderer.create(<RegularPage />).toJSON();
    expect(page).toMatchSnapshot();
  });
});
