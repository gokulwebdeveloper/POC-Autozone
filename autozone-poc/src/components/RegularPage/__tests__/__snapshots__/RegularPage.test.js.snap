// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test CheckoutPage SnapShot Match 1`] = `
<div
  className="container-fluid main-container"
>
  <div
    className="row header-row"
  >
    <img
      className="img-fluid w-100 header-image"
      src="bg-8.jpg"
    />
  </div>
  <div
    className="container justify-content-center my-5 content-container"
  >
    <div
      className="row content-row"
    >
      <h2
        className="mb-5"
      >
        Vivamus sed nunc in arcu cursus mollis quis et orci. Interdum et malesuada
      </h2>
      <p
        className="mb-5"
      >
        Mauris viverra cursus ante laoreet eleifend. Donec vel fringilla ante. Aenean finibus velit id urna vehicula, nec maximus est sollicitudin. Praesent at tempus lectus, eleifend blandit felis. Fusce augue arcu, consequat a nisl aliquet, consectetur elementum turpis. Donec iaculis lobortis nisl, et viverra risus imperdiet eu. Etiam mollis posuere elit non sagittis. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc quis arcu a magna sodales venenatis. Integer non diam sit amet magna luctus mollis ac eu nisi. In accumsan tellus ut dapibus blandit.
      </p>
      <blockquote
        className="mb-5 quote-blockquote"
      >
        <h6
          className="d-flex quote-header"
        >
          <i
            aria-hidden="true"
            className="fa fa-quote-left mr-4 quote-icon"
          />
          Quisque sagittis non ex eget vestibulum. Sed nec ultrices dui. Cras et sagittis erat. Maecenas pulvinar, turpis in dictum tincidunt, dolor nibh lacinia lacus.
        </h6>
        <span
          className="ml-5"
        >
          Liam Neeson
        </span>
      </blockquote>
      <p>
        Praesent ac magna sed massa euismod congue vitae vitae risus. Nulla lorem augue, mollis non est et, eleifend elementum ante. Nunc id pharetra magna. Praesent vel orci ornare, blandit mi sed, aliquet nisi. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos.
      </p>
    </div>
  </div>
</div>
`;
